test/testText.cpp

```diff
@@ -164,4 +200,116 @@ TEST_CASE("Text with composite glyphs", "[tvgText]")
     Initializer::term();
 }
 
+TEST_CASE("Text Bounding Box", "[tvgText]")
+{
+    // Expected values measured from Arial 32pt font rendering
+    // These are used as regression test values.
+
+    REQUIRE(Initializer::init() == Result::Success);
+    {
+        auto canvas = unique_ptr<SwCanvas>(SwCanvas::gen());
+        uint32_t buffer[100 * 100];
+        REQUIRE(canvas->target(buffer, 100, 100, 100, ColorSpace::ARGB8888) == Result::Success);
+
+        REQUIRE(Text::load(TEST_DIR"/Arial.ttf") == Result::Success);
+
+        auto text = Text::gen();
+        REQUIRE(canvas->push(text) == Result::Success);
+        REQUIRE(canvas->sync() == Result::Success);
+
+        //Negative
+        float x = 0, y = 0, w = 0, h = 0;
+        REQUIRE(text->bounds(&x, &y, &w, &h) == Result::InsufficientCondition);
+
+        //Case 1
+        REQUIRE(text->font("Arial", 32) == Result::Success);
+        REQUIRE(text->text("TEST") == Result::Success);
+        REQUIRE(text->translate(100.0f, 111.0f) == Result::Success);
+        REQUIRE(text->bounds(&x, &y, &w, &h) == Result::Success);
+
+        REQUIRE(x == 101.0f);
+        REQUIRE(y == 118.5625f);
+        REQUIRE(w == 107.1875f);
+        REQUIRE(h == Approx(31.58334f).margin(0.001f));
+
+        REQUIRE(canvas->update() == Result::Success);
+        Point pts[4];
+        REQUIRE(text->bounds(pts) == Result::Success);
+        REQUIRE(pts[0].x == 101.0f);
+        REQUIRE(pts[3].x == 101.0f);
+        REQUIRE(pts[1].x == 208.1875f);
+        REQUIRE(pts[2].x == 208.1875f);
+        REQUIRE(pts[0].y == 118.5625f);
+        REQUIRE(pts[1].y == 118.5625f);
+        REQUIRE(pts[2].y == 150.14584f);
+        REQUIRE(pts[3].y == 150.14584f);
+
+        //Case 2
+        REQUIRE(text->text("BOUNDS") == Result::Success);
+        auto identity = Matrix{1.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 1.0f};
+        REQUIRE(text->transform(identity) == Result::Success);
+        REQUIRE(text->bounds(&x, &y, &w, &h) == Result::Success);
+
+        REQUIRE(x == 3.125f);
+        REQUIRE(y == Approx(7.54167f).margin(0.001f));
+        REQUIRE(w == 177.1875f);
+        REQUIRE(h == Approx(31.60417f).margin(0.001f));
+
+        REQUIRE(canvas->update() == Result::Success);
+        REQUIRE(text->bounds(pts) == Result::Success);
+        REQUIRE(pts[0].x == 3.125f);
+        REQUIRE(pts[3].x == 3.125f);
+        REQUIRE(pts[1].x == 180.3125f);
+        REQUIRE(pts[2].x == 180.3125f);
+        REQUIRE(pts[0].y == Approx(7.54167f).margin(0.001f));
+        REQUIRE(pts[1].y == Approx(7.54167f).margin(0.001f));
+        REQUIRE(pts[2].y == Approx(39.14584f).margin(0.001f));
+        REQUIRE(pts[3].y == Approx(39.14584f).margin(0.001f));
+    }
+    REQUIRE(Initializer::term() == Result::Success);
+}
```

Comment:
Is this kind of test still considered valid?
The expected values are statically determined, but int this test, I used the values returned by the current function

Reply:
looks acceptable.